src/io/unix/sync.rs

```diff
@@ -0,0 +1,197 @@
+// Sync linux
+
+use crate::prelude::*;
+use bytes::{Bytes, BytesMut};
+use nix::errno::Errno;
+use nix::fcntl::FcntlArg;
+use std::sync::atomic::AtomicU64;
+use std::{
+    os::unix::prelude::{AsRawFd, FileExt},
+    time::SystemTime,
+};
+use tokio::fs::OpenOptions;
+
+/// IO driver for file operations
+#[derive(Debug, Clone, Default)]
+pub struct IODriver;
+
+impl IODriver {
+    pub(crate) async fn open(&self, path: impl AsRef<Path>) -> IOResult<File> {
+        File::from_file(path, |f| f.create(false).append(true).read(true)).await
+    }
+
+    pub(crate) async fn create(&self, path: impl AsRef<Path>) -> IOResult<File> {
+        File::from_file(path, |f| f.create(true).write(true).read(true)).await
+    }
+}
+
+// This size was selected based on results of tests
+// These tests were performed on 8-core machine with HDD
+// Results are confirmed on 8 and 80 threads, for put and get
+const MAX_SYNC_OPERATION_SIZE: usize = 81_920;
+
+#[derive(Debug, Clone)]
+pub(crate) struct File {
+    pub(super) no_lock_fd: Arc<StdFile>,
+    pub(super) size: Arc<AtomicU64>,
+}
+
+#[derive(PartialEq, Eq)]
+enum LockAcquisitionResult {
+    Acquired,
+    AlreadyLocked,
+    Error(Errno),
+}
+
+impl File {
+    pub(crate) fn size(&self) -> u64 {
+        self.size.load(ORD)
```

Comment:
In other places `Ordering::SeqCst` is used, so it is better to use it here too explicitly